src/http.cc

```diff
@@ -1393,6 +1395,16 @@ HttpStateData::truncateVirginBody()
     }
 }
 
+/// called when an incomplete reply body preliminary ends with unexpected EOF
+void
+HttpStateData::failOnPrematureReplyBodyEof()
+{
+    const auto err = new ErrorState(ERR_READ_ERROR, Http::scBadGateway, fwd->request, fwd->al);
+    static const auto d = MakeNamedErrorDetail("SRV_PREMATURE_EOF");
+    err->detailError(d);
+    fwd->fail(err);
+}
+
```

Comment:
It looks like that false dataWritten means no received bytes at all (since any non-zero read where aBufferLength>0 sets dataWritten to true). I guess we need ERR_ZERO_SIZE_OBJECT here then.

Reply:
> false dataWritten means no received bytes at all

That is my recollection as well, but I did not double check it.

> I guess we need ERR_ZERO_SIZE_OBJECT here then.

Yes, let's use that error code.

Reply:
Ok.